### PR TITLE
[OPIK-294] Add last created experiment at to dataset table

### PR DESCRIPTION
**Pull request by thiagohora**
## Details

Migration to add field last created experiment at which will allow us to filter by datasets with experiments and sort by it.

## Issues
OPIK-294

### CODE REVIEW

**Review by idoberko2 — APPROVED** [review]
LGTM

File: apps/opik-backend/src/main/resources/liquibase/db-app-state/migrations/000003_add_last_created_experiment_at_to_datasets_table.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+--liquibase formatted sql
+--changeset thiagohora:add_last_created_experiment_at_to_datasets_table
+
+ALTER TABLE datasets ADD COLUMN last_created_experiment_at TIMESTAMP(6) DEFAULT NULL;
```
Comment (andrescrz) [inline apps/opik-backend/src/main/resources/liquibase/db-app-state/migrations/000003_add_last_created_experiment_at_to_datasets_table.sql:4]: Minor: with idempotency would be great e.g: ALTER TABLE IF EXITS... and ADD COLUMN IF NOT EXISTS. But I believe not supported by MySQL.

**Review by andrescrz — APPROVED** [review]
Just left a thought, but believe not possible in MySQL. This is good to go.

File: apps/opik-backend/src/main/resources/liquibase/db-app-state/migrations/000003_add_last_created_experiment_at_to_datasets_table.sql (hunk 1 above)
Comment (thiagohora) [reply apps/opik-backend/src/main/resources/liquibase/db-app-state/migrations/000003_add_last_created_experiment_at_to_datasets_table.sql:4]: Yeah, unfortunately, MySQL doesn't support it yet. Only `MariaDB` does :( 

REF: https://dev.mysql.com/doc/refman/8.4/en/alter-table.html

**Pull request merged**